modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -228,15 +230,34 @@ const DatePicker = ({
 				}
 			}
 			else if (initialValueMemoized) {
-				inputRef.current.value = moment(initialValueMemoized).format(
-					dateMask.toUpperCase()
+				var year = parseInt(
+					initialValueMemoized.substr(YEARS_INDEX),
+					10
 				);
+
+				if (year > 50 && year < 100) {
+					inputRef.current.value = moment(initialValueMemoized)
+						.subtract(1900, 'years')
+						.format(dateMask.toUpperCase());
+				}
+				else if (year <= 50) {
+					inputRef.current.value = moment(initialValueMemoized)
+						.subtract(2000, 'years')
+						.format(dateMask.toUpperCase());
+				}
+				else {
+					inputRef.current.value = moment(
+						initialValueMemoized
+					).format(dateMask.toUpperCase());
+				}
```

Comment:
I suggest you to change the order the `if`s occur and the way you  set the `inputRef` to simplify the code and increase the code reuse:
```suggestion
				const date = moment(initialValueMemoized);
				
				if (year <= 50) {
					date.subtract(1900, 'years');
				}
				else if (year < 100) {
					date.subtract(2000, 'years');
				}
				
				inputRef.current.value = date.format(dateMask.toUpperCase());
```